### ISSUE TITLE
Testing with addHTTP2Upgrade

### ISSUE THREAD

**Issue by davidjohannesbotha** [hummingbird-project/hummingbird-examples#41]
Hi all,

This might not actually be a bug, but the documentation is not complete, so I thought I would post it here. My HB server is using the addHTTP2Upgrade, and all works in the 'production' settting. However, with the addHTTP2Upgrade, for all tests (even the simplest endpoints such as: 
```swift
self.router.get("/") { _ in
            return "Hello"
        }
``` 
the test fails.  The tests succeed without the addHTTP2Upgrade, but I would really like to test all my endpoints with this enabled.  The test fails with the following log message: `/Library/Developer/Xcode/DerivedData/HummingbirdServer-cqscixbrudpxqncfbsfmyuaasefc/SourcePackages/checkouts/swift-nio/Sources/NIOCore/EventLoopFuture.swift:1284: error: -[AppTests.AppTests testCreate] : failed: caught error: "alreadyClosed"`. 

Once again, sorry if this is not the place mention this issue.

Kind regards
David

**Comment by adam-fowler** [comment]
Hmm not a totally helpful error message. 

The HummingbirdXCT test client doesn't support HTTP2. To test it you would have to use [AsyncHTTPClient](https://github.com/swift-server/async-http-client) which does support HTTP2. And then connect with a TLSConfiguration where `certificateVerification` is set to either `.none` or `.noHostnameVerification`.

**Comment by adam-fowler** [comment]
So I created this to test HTTP2 and TLS https://github.com/hummingbird-project/hummingbird-xct-async-http-client

And found shutdown of HTTP2 isn't working too great. I need to talk the swift-nio guys to find out the best way to resolve this.

**Comment by adam-fowler** [comment]
Fixed shutdown of HTTP2 in v1.6 of HummingbirdCore
